pkg/scheduler/scheduler.go

```diff
@@ -156,6 +156,17 @@ func (s *PubSubScheduler) Schedule(campaign db.Campaign) error {
 }
 
 func (s *PubSubScheduler) publishTask(ctx context.Context, task *db.Task) error {
+
+	//Check if task.CampaignID belongs to a cancelled/halted Campaign. If so skip it.
+	taskIsCancelled, err := s.db.IsCampaignCancelled(task.CampaignID)
```

Comment:
This function will be called a lot, so we'll need to make sure its performant. I think the change suggested in `pkg/db/client.go` should help, but it might be a good idea for us to collect some performance data on this operation in a real database setup.

Reply:
Another thing to consider here is cost. Will repeated database calls here blow up the cost of the project? Should we replicate campaign status to the cache for performance/cost?

Reply:
Yeah, definitely something we need to watch. Let's keep an eye on this the next time that a proper campaign is launched and see if it causes a noticeable difference. I'm hesitant to add some complexity / opportunity for the cache to be misaligned with the db for something we're not sure is an issue yet. I'd probably ask to move this to a future ticket as long as it's not totally derailing usability.